components/support/nimbus-fml/src/parser.rs

```diff
@@ -69,6 +69,69 @@ pub(crate) struct ManifestFrontEnd {
     channels: Vec<String>,
 }
 
+trait MergableDefault: Sized {
+    fn merge(self, new: Self) -> Self;
+}
+
+impl MergableDefault for serde_json::Map<String, serde_json::Value> {
+    fn merge(self, new: Self) -> Self {
+        let mut merged = serde_json::Map::new();
+        for (key, val) in self {
+            merged.insert(key, val);
+        }
+        for (key, val) in new {
+            if merged.contains_key(&key) {
+                let old_val = merged.get(&key).unwrap().clone();
+                merged.insert(key, old_val.merge(val));
+            } else {
+                merged.insert(key, val);
+            }
+        }
+        merged
+    }
+}
+
+impl MergableDefault for serde_json::Value {
+    fn merge(self, new: Self) -> Self {
+        use serde_json::Value::{Array, Number, Object, String};
+        match (self, new) {
+            (Number(old), Number(new)) => Number(old.merge(new)),
+            (String(old), String(new)) => String(old.merge(new)),
+            (Array(old), Array(new)) => Array(old.merge(new)),
+            (Object(old), Object(new)) => Object(old.merge(new)),
+            (_, new) => new,
```

Comment:
note to myself that I need to think more about this if there are any more interesting cases